### PR TITLE
Migrations interface

### PR DESCRIPTION
**Pull request by melnychukvitaliy**
Here is an example how we can implement our own interface with to manage migrations

it doesn't integrations with other libraries but we can think about helpers to convert migrations into `yml` files and then simply use `crispy-cli` to sync with serrver

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,6 +14,38 @@ CLI to communicate with `crispy-api`.
...
+
+```
+Usage: crispy-cli migrations [options] [command]
+
+Options:
+  -h, --help               output usage information
+
+Commands:
+  up <migrationName>       Runs the specified migration
+  down <migrationName>     Will undo the specified migration
```
Comment (xTNTx) [inline README.md]: I propose to use semantic verbs like `migrate`/`rollback` etc.

File: README.md [hunk 2]
```diff
@@ -14,6 +14,38 @@ CLI to communicate with `crispy-api`.
...
+  publish <migrationName>  Send specified migration to server
+  create <migrationName>   Create new migration
+  sync                     Sync migrations directory with server
```
Comment (xTNTx) [inline README.md]: What is the difference between `sync` and `publish`?

File: examples/create-and-alter-table/migrations/1-create-users-table.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    id INT(6) UNSIGNED AUTO_INCREMENT PRIMARY KEY,
+    firstname VARCHAR(30) NOT NULL,
+    lastname VARCHAR(30) NOT NULL,
+    email VARCHAR(50),
+    reg_date TIMESTAMP
+  )
+
+down:
+  table: users
+  operation: delete
```
Comment (xTNTx) [inline examples/create-and-alter-table/migrations/1-create-users-table.yml]: Might be for later, but what is the list of possible operations? The query is `DROP` but the operation is `delete`?

File: README.md (hunk 2 above)
Comment (melnychukvitaliy) [reply README.md]: the intention was to split add particular migration, and sync command (where migrations can be deleted or added). `sync` Command will be good for CI, where you have to add only one command

also, I think `unpublish` command is missed here.
Generally, we will have `publish`, `unpublish` to manage migration, but `migrationName` is required there

File: examples/create-and-alter-table/migrations/1-create-users-table.yml (hunk 3 above)
Comment (melnychukvitaliy) [reply examples/create-and-alter-table/migrations/1-create-users-table.yml]: Yeah, sent it as an example. 
attached issue for it

File: README.md (hunk 1 above)
Comment (melnychukvitaliy) [reply README.md]: I have experienced with few schema migrations and they have `up`, `down`

but, generally, like this idea 👍

File: examples/create-and-alter-table/migrations/1-create-users-table.yml [hunk 4]
```diff
@@ -0,0 +1,22 @@
+version: 0.0.1
```
[the remark this answers is not in the document]
Comment (melnychukvitaliy) [reply examples/create-and-alter-table/migrations/1-create-users-table.yml]: I think we have to follow migration version in the name of migration, as all migration schemas do

here, I've added crispy-cli version of syntax, to have backward compatibility. 
We can skip it, as it's our first release and add it once we need

[the remark this answers is not in the document]
Comment (xTNTx) [reply examples/create-and-alter-table/migrations/1-create-users-table.yml]: It's ok to keep it, just wanted to confirm my understanding.

File: README.md (hunk 2 above)
Comment (xTNTx) [reply README.md]: I think we should talk more about usage scenarios for these commands. When and how often a user would use `sync`, `publish`, `unpublish`. Maybe we will able to merge these two into one (just a raw theory), e.g. if migration name is not passed - sync all? 

We may pull out theses command from the current PR to not block it.

Comment (melnychukvitaliy) [reply README.md]: Yeah, we might start from `sync` command a then try to add more commands based on scenarios we have.

Basically, we are able to start from `migrate/rollback` command as we definitely this functionality.
then start with `sync` command along with API implementation for it

**Pull request merged**